x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/fielddata/ScriptLongFieldData.java

```diff
@@ -0,0 +1,171 @@
+import org.elasticsearch.index.fielddata.IndexFieldDataCache;
+import org.elasticsearch.index.fielddata.IndexNumericFieldData;
+import org.elasticsearch.index.fielddata.LeafNumericFieldData;
+import org.elasticsearch.index.fielddata.ScriptDocValues;
+import org.elasticsearch.index.fielddata.SearchLookupAware;
+import org.elasticsearch.index.fielddata.SortedBinaryDocValues;
+import org.elasticsearch.index.fielddata.SortedNumericDoubleValues;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.search.aggregations.support.CoreValuesSourceType;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xpack.runtimefields.LongScriptFieldScript;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class ScriptLongFieldData extends IndexNumericFieldData implements SearchLookupAware {
+
+    public static class Builder implements IndexFieldData.Builder {
+
+        private final LongScriptFieldScript.Factory scriptFactory;
+
+        public Builder(LongScriptFieldScript.Factory scriptFactory) {
+            this.scriptFactory = scriptFactory;
+        }
+
+        @Override
+        public ScriptLongFieldData build(
+            IndexSettings indexSettings,
+            MappedFieldType fieldType,
+            IndexFieldDataCache cache,
+            CircuitBreakerService breakerService,
+            MapperService mapperService
+        ) {
+            return new ScriptLongFieldData(indexSettings.getIndex(), fieldType.name(), scriptFactory);
+        }
+    }
+
+    private final Index index;
+    private final String fieldName;
+    private final LongScriptFieldScript.Factory scriptFactory;
+    private final SetOnce<LongScriptFieldScript.LeafFactory> leafFactory = new SetOnce<>();
+
+    private ScriptLongFieldData(Index index, String fieldName, LongScriptFieldScript.Factory scriptFactory) {
+        this.index = index;
+        this.fieldName = fieldName;
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    public void setSearchLookup(SearchLookup searchLookup) {
+        // TODO wire the params from the mappings definition, we don't parse them yet
+        this.leafFactory.set(scriptFactory.newFactory(Collections.emptyMap(), searchLookup));
+    }
+
+    @Override
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    @Override
+    public ValuesSourceType getValuesSourceType() {
+        return CoreValuesSourceType.NUMERIC;
+    }
+
+    @Override
+    public ScriptLongLeafFieldData load(LeafReaderContext context) {
+        try {
+            return loadDirect(context);
+        } catch (Exception e) {
+            if (e instanceof ElasticsearchException) {
```

Comment:
another change that I lost somehow, but this should rather be ExceptionsHelper#convertToElastic . Can you make the same change in the binary one too?

Reply:
:+1: